pkg/ice/candidate.go

```diff
@@ -104,20 +102,12 @@ func (c *Candidate) recvLoop() {
 			return
 		}
 
-		if stun.IsSTUN(buffer[:n]) {
-			m, err := stun.NewMessage(buffer[:n])
-			if err != nil {
-				iceLog.Warnf("Failed to handle decode ICE from %s to %s: %v", c.addr(), srcAddr, err)
-				continue
-			}
-			err = c.agent.run(func(agent *Agent) {
-				agent.handleInbound(m, c, srcAddr)
-			})
-			if err != nil {
-				iceLog.Warnf("Failed to handle message: %v", err)
-			}
-
-			continue
+		m := buffer[:n]
+		err = c.agent.run(func(agent *Agent) {
+			agent.handleInbound(m, c, srcAddr)
```

Comment:
There is a known problem here: `run` is executed asynchronously. This means the buffer will be used concurrently in the `recvLoop` and the code executed by `run`. This can cause corruption of the buffer. This change makes this problem worse since it now occurs with every package, not just the STUN packages.
We intend to solve the original problem by making a copy of the buffer before passing it to the `run` method. However, this solution doesn't seem appropriate for this case since we would have to make a copy for all incoming data and the copy would mostly not be used. I suggest to split this up again: have a separate call to `run` for user data that finds the right remote candidate and marks it as seen.